cmd/system/powershell.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) arkade author(s) 2024. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package system
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/alexellis/arkade/pkg/archive"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallPowershell() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "pwsh",
+		Short: "Install Powershell",
+		Long:  `Install Powershell cross-platform task automation solution.`,
+		Example: `  arkade system install pwsh
+  arkade system install pwsh --version <version>`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("version", "v", "", "The version or leave blank to determine the latest available version")
+	command.Flags().String("path", "/opt/microsoft/powershell", "Installation path, where a powershell subfolder will be created")
+	command.Flags().Bool("progress", true, "Show download progress")
+	command.Flags().String("arch", "", "CPU architecture i.e. amd64")
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+
+		arch, osVer := env.GetClientArch()
+
+		if strings.ToLower(osVer) != "linux" {
+			return fmt.Errorf("this app only supports Linux")
+		}
+
+		if cmd.Flags().Changed("arch") {
+			arch, _ = cmd.Flags().GetString("arch")
+		}
+
+		dlArch := arch
+		if arch == "x86_64" {
+			dlArch = "x64"
+		} else if arch == "aarch64" {
+			dlArch = "arm64"
+		} else if arch == "armv7" || arch == "armv7l" {
+			dlArch = "arm32"
+		}
+
+		if version == "" {
+			v, err := get.FindGitHubRelease("PowerShell", "PowerShell")
+			if err != nil {
+				return err
+			}
+			version = v
+		} else if !strings.HasPrefix(version, "v") {
+			version = "v" + version
+		}
+
+		fmt.Printf("Installing version: %s for: %s\n", version, dlArch)
+
+		semVer := strings.TrimPrefix(version, "v")
```

Comment:
This package may cover edge cases better: https://github.com/Masterminds/semver

Reply:
Done